_posts/2020-12-17-solidity-v0.8.0-release-announcement.md

```diff
@@ -0,0 +1,237 @@
+backwards-incompatible adjustment in syntax or semantics. For a detailed explanation,
+please see the [documentation](https://docs.soliditylang.org/en/latest/080-breaking-changes.html).
+
+The change that will affect most users is that arithmetic operations are now checked by default,
+which means that overflow and underflow will cause a revert. This feature can be disabled locally.
+
+A second change that is very visible is that the ABI coder v2 is activated by default. You can
+activate the old coder using ``pragma abicoder v1``, or explicitly select v2 using ``pragma abicoder v2``.
+ABI coder v2 is more complex but also performs more checks on the input and supports more types.
+
+Furthermore, internal errors like division by zero, assertion failure and others
+do not use the invalid opcode anymore, but use ``revert`` with a special error message,
+in order to not waste gas in such situations.
+
+Another important change is that we restricted the possibilities of explicit conversions
+to avoid ambiguities. All conversions that were possible before are still possible, but
+you might have to perform two conversions to get there - which does not affect the generated code, though.
+
+These are only some of the more important breaking changes in 0.8.0., please read the changelog below for the full list!
+
+### Checked Arithmetic
+
+The "Checked Arithmetic" feature of Solidity 0.8.x consists of three sub-features:
+- Revert on assertion failures and similar conditions instead of using the invalid opcode.
+- Actual checked arithmetic.
+- ``unchecked`` blocks.
+
+#### Revert on assertion failures and similar conditions instead of using the invalid opcode
+
+Prevously, internal errors like division by zero, failing assertions, array access out of bounds,
+etc., would result in an invalid opcode being executed. The idea was to distinguish
+these more critical errors from non-critical errors that lie outside of the domain
+of the Smart Contract programmer like invalid input, not enough balance for a token transfer,
+and so on. These non-critical errors use the ``revert`` opcode and optionally add an error description.
+
+The problem with the invalid opcode is that, in contrast to the ``revert`` opcode, it consumes all gas available.
+This makes it very expensive and you should try to avoid it at all cost.
+
+We wanted to consider arithmetic overflow as a critical error, but did not want to cause it to consume all gas.
+As a middle ground, we chose to use the ``revert`` opcode, but provide a different error data
+so that static (and dynamic) analysis tools can easily distinguish them.
+
+More specifically:
+ - Non-critical errors will revert either with empty error data or with error data that
+   corresponds to the ABI-encoding of a function call to a function with the signature ``Error(string)``.
+ - Critical errors revert with error data that corresponds to the ABI-encoding of a function call
+   to a function with the signature ``Panic(uint256)``.
```

Comment:
I see that github renders the list correctly in this case but indenting top-level list items sometimes results in a verbatim block instead so it's better to avoid it.

```suggestion
- Non-critical errors will revert either with empty error data or with error data that
    corresponds to the ABI-encoding of a function call to a function with the signature ``Error(string)``.
- Critical errors revert with error data that corresponds to the ABI-encoding of a function call
    to a function with the signature ``Panic(uint256)``.
```

Some lists below also have this extra indent.